client/layout/guided-tours/index.js

```diff
@@ -15,14 +15,20 @@ import QueryPreferences from 'components/data/query-preferences';
 import RootChild from 'components/root-child';
 import { getGuidedTourState } from 'state/ui/guided-tours/selectors';
 import { getLastAction } from 'state/ui/action-log/selectors';
-import { getSectionName, isSectionLoading } from 'state/ui/selectors';
-import { nextGuidedTourStep, quitGuidedTour } from 'state/ui/guided-tours/actions';
+import { getInitialQueryArguments, getSectionName, isSectionLoading } from 'state/ui/selectors';
+import { nextGuidedTourStep, quitGuidedTour, resetGuidedToursHistory } from 'state/ui/guided-tours/actions';
 
 class GuidedTours extends Component {
 	shouldComponentUpdate( nextProps ) {
 		return this.props.tourState !== nextProps.tourState;
 	}
 
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.requestedTour === 'reset' && this.props.requestedTour !== 'reset' ) {
```

Comment:
Given this is handling the `?tour=reset` param, this seems fair. I feel like there could be a better place for this or a better approach but I think this might just be because I'm not too familiar with GT.